### DISCUSSION TITLE
Classification of mutations within mmpL5 vs. mmpS5 vs. mmpR loci

### DISCUSSION THREAD

**Discussion by frogtraveler** [jodyphelan/TBProfiler#303] [Q&A] [answered]
Hi!

Could you please help me to understand how the decision is made to which of the 3 overlapping loci to assign a mutation: mmpL5 vs. mmpS5 vs. mmpR? E.g. mutation in genome position 778908C>G is not in WHO catalogue or TBDB. It can be classified as mmpR (Rv0678) c.-82C>G or as mmpL5 c.-428G>C. The TB Profiler picks the later but it seems like the first one would be more significant in terms of potential effect on resistance.
 
Or another example, mutation mmpR (Rv0678) c.-11C>A is listed in WHO as "uncertain", but TB Profiler identifies it as mmpS5 c.-74G>T, also with "uncertain" confidence in TBDB. Why was decision made for mmpS5 nomenclature in that case? Wouldn't the mutation in promoter region closer to the start codon more likely have an effect? 

I was just thinking that especially for mutations not found in WHO & TBDB, it might be helpful for a user to know the mutation position in relation to a gene where it is most likely to have an effect.

Thank you!
Varvara

**Comment by jodyphelan** [comment]
Hi @frogtraveler ,

Which version are you using?

The most relevant mutation is chosen based on the following criteria (in order of importance):
1. If any of the changes is associated with drug resistance
2. Ranking based on the type of variant for example: frameshift > missense > promoter mutation > synonymous
3. The first change with a WHO annotation (mmpS5 c.-74G>T falls into this rule)

If none of these are fulfilled (like 778908C>G as they are all classed as promoter mutations), then it will just pick the first change based on the ordering of the gene in the genome. This is why mmpL5 c.-428G>C has been chosen here.

There are a few solutions that I can think of:
1. We give the genes an importance value which could be used to rank and pick the changes
2. We add an option to report all changes in the text output
3. You can use the json reports as they do store all consequences for example:

```
{
  "chrom": "Chromosome",
  "genome_pos": 778908,
  "ref": "C",
  "alt": "G",
  "depth": 100,
  "freq": 1,
  "forward_reads": null,
  "reverse_reads": null,
  "sv": false,
  "sv_len": null,
  "feature_id": "CCP43419",
  "type": "upstream_gene_variant",
  "nucleotide_change": "c.-428G>C",
  "protein_change": "",
  "change": "c.-428G>C",
  "locus_tag": "Rv0676c",
  "gene": "mmpL5",
  "gene_associated_drugs": [
    "clofazimine",
    "bedaquiline"
  ],
  "alternate_consequences": [
    {
      "gene_name": "mmpS5",
      "gene_id": "Rv0677c",
      "feature_id": "CCP43420",
      "type": "upstream_gene_variant",
      "nucleotide_change": "c.-3G>C",
      "protein_change": ""
    },
    {
      "gene_name": "Rv0678",
      "gene_id": "Rv0678",
      "feature_id": "CCP43421",
      "type": "upstream_gene_variant",
      "nucleotide_change": "c.-82C>G",
      "protein_change": ""
    }
  ]
}
```

If you have a custom way of reporting based on json file you could pull out all consequences from here.

Let me know your thoughts!

**Reply by frogtraveler** [reply]
Thank you for your reply, Jody! 
We are using v.4.4.2.
Ranking first based on known association with resistance and then based on the type of variant makes total sense. I suppose it would be nice if close-proximity promoter mutations were weighted more than distant promoter; e.g. frameshift > missense > promoter mutation within 35 bp from start codon > promoter mutation further than 35 bp from start codon > synonymous. Or smth like that.
It seems to me that giving genes the importance values would be most helpful. I assume reporting mutations preferentially in Rv0678 as the most prevalent mechanism of resistance to BDQ would make sense. Is there a way then to build in a check for possible epistatic interaction between Rv0678 and LoF in mmpS5 and mmpL5?
Thank you!
Varvara

**Reply by jodyphelan** [reply, marked as answer]
> I suppose it would be nice if close-proximity promoter mutations were weighted more than distant promoter; e.g. frameshift > missense > promoter mutation within 35 bp from start codon > promoter mutation further than 35 bp from start codon > synonymous. Or smth like that.

I think this should be relatively simple to implement. Ill get than into the next version

> It seems to me that giving genes the importance values would be most helpful. I assume reporting mutations preferentially in Rv0678 as the most prevalent mechanism of resistance to BDQ would make sense.

Will look into this!

> Is there a way then to build in a check for possible epistatic interaction between Rv0678 and LoF in mmpS5 and mmpL5?

I've been working on getting rules like this into the logic. The logic is already built into tb-profiler. Just need to add it to tbdb so that they automatically load. Will let you know when this is ready.

**Reply by frogtraveler** [reply]
Looking forward to the updates! The expert rules would be super helpful! We had to build some of our own to further parse TB Profiler output, but it would be great if those were built-in (yet modular, so in case some become outdated and need to be updated, we could easily comment specific rules out). Just thinking out loud :)

**Reply by jodyphelan** [reply]
Good to hear it will be useful :)  The exact syntax still needs to be ironed out but will keep you updated!
